include/simdjson/haswell/bitmanipulation.h

```diff
@@ -50,7 +50,7 @@ simdjson_really_inline bool add_overflow(uint64_t value1, uint64_t value2,
                        reinterpret_cast<unsigned __int64 *>(result));
 #else
   return __builtin_uaddll_overflow(value1, value2,
-                                   (unsigned long long *)result);
+                                   reinterpret_cast<unsigned long long *>(result));
```

Comment:
Are you sure that this should be a reinterpret_cast?

Reply:
At first glance it looks like it can be static_cast because uint64_t and unsigned long long int seems to be
both unsigned int's with 8 byte in size on 64 bit systems.

But the standard says:

signed char ≤ short int ≤ int ≤ long int ≤ long long int

and uint64_t is internally 
unsigned long int
vs.
unsigned long long int

If you only look at this function parameters all are pointers to uint64_t integer types.
From my point of view the correct function to use here is:
__builtin_uaddl_overflow (with single L)
and not
__builtin_uaddll_overflow (with double L)
then no cast is needed

Because i haven't digged deeper into the code and i don't want make insecure code changes so i think in this case reinterpret_cast is correct.

Have a look at the small example snippet tested with gcc9 and gcc10.

int main()
{
    uint64_t val= 1234;
    uint64_t *p = &val;
    unsigned long int *p1 = p; // okay
    unsigned long long int *p2 = p;  // won't compile
    unsigned long long int *p3 = p1;  // won't compile

    unsigned long int* ulli1= static_cast<unsigned long int*>(p); // okay
    unsigned long long int* ulli2= reinterpret_cast<unsigned long long int*>(p); //okay
    unsigned long long int* ulli3= static_cast<unsigned long long int*>(p); // won't compile

    auto s1= sizeof (unsigned long long int); // 8 bytes on my machine
    auto s2= sizeof (unsigned long int); // 8 bytes on my machine
    
    return 0;
}

Reply:
We can handle this as a separate PR.